rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -323,21 +334,28 @@ def required_packages(cls) -> List[Text]:
 
     def __init__(
         self,
-        component_config: Optional[Dict[Text, Any]] = None,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
```

Comment:
I wonder if there's a good way to get rid of the extra kwargs in a reasonable way. I mean those are only used in `load`. So we could move that there / o a separate function that might include more checks regarding "does the config we're using work with the model that we loaded" ? ( e.g. config might say embedding dim x but our loaded model has dim y then we know something went really really wrong ) Maybe that is done elsewhere ... but not having those extra args here would look nice / feel cleaner to me. What do you think?

Reply:
You mean `**kwargs: Any,`? 
This is needed in-case we want an output from a parent node to be used in the loading of a child node.

Reply:
Not really, more like in no extra arguments in the intialisation, which should all either be None or set to something, and a separate function for initialising the DIET classifier using the loaded resources. 🤔  
.... Sorry, I realize how that there's no benefit in that if don't have those extra checks. That would be just one init and the four lines of "overwrite this with that". Please just ignore my spam here ^^.